src/Mex.cpp

```diff
@@ -0,0 +1,159 @@
+#include "plugin.hpp"
+#include "Muxlicer.hpp"
+
+
+struct Mex : Module {
+	static const int numSteps = 8;
+	enum ParamIds {
+		ENUMS(STEP_PARAM, numSteps),
+		NUM_PARAMS
+	};
+	enum InputIds {
+		GATE_IN_INPUT,
+		NUM_INPUTS
+	};
+	enum OutputIds {
+		OUT_OUTPUT,
+		NUM_OUTPUTS
+	};
+	enum LightIds {
+		ENUMS(LED, numSteps),
+		NUM_LIGHTS
+	};
+	enum StepState {
+		GATE_IN_MODE,
+		MUTE_MODE,
+		MUXLICER_MODE
+	};
+
+	dsp::SchmittTrigger gateInTrigger;
+
+	// this expander communicates with the mother module (Muxlicer) purely
+	// through this pointer (it cannot modify Muxlicer, read-only)
+	Muxlicer const* mother = nullptr;
```

Comment:
I can't see how this *doesn't* crash Rack when Muxlicer is removed. If you have a chain of 2 or more Mex's, if Muxlicer is removed, it will take 2 frames for the pointer to be gone from the Mex's. If the Mex tries to access it, it will segfault.
Expanders should pass messages of data, not pointers that might be freed.

Reply:
I see what you mean. It's weird though, I can stack 20 Mexs set multithreading and delete the host Muxlicer without issue. But if can't explain why then it's a bit worrying. I have a partially finished / more complicated implemented with message passing that I can revert to if needed?

Reply:
Proper message passing would be best.

If you need direct access to the "mother" module, you can obtain it by following the path of expanders in a recusive call. But if you store pointers across `process()` calls, it may result in undefined behavior when the module is deleted and you try to access it.

Reply:
I've implemented the recursive version (as it's simpler), but still happy to revert if needed (though I don't understand the message parsing as well).

Reply:
That'll be fine